LSPS4/README.md

```diff
@@ -0,0 +1,1923 @@
+it will be unable to claim the funds.
+The client can protect against this (i.e. client does not trust the LSP) by
+simply not providing the preimage until it sees the funding transaction has
+been broadcast and is in its own local mempool.
+
+If neither the client nor the LSP trust each other, they will deadlock: the
+untrusting LSP withholds the funding transaction until it sees the payment
+preimage, the untrusting client withholds the preimage until it sees the
+funding transaction.
+
+In an "LSP trusts client" model, the client MAY send the preimage
+immediately, or MAY wait for the funding transaction to be broadcast and
+validated as being correct before sending the preimage, or MAY wait for 1
+or more confirmations before sending the preimage.
+
+As the client is allowed to defer preimage sending until there is no
+need to trust the LSP (i.e. the funding transaction is deeply confirmed
+enough that reorganization is negligible), the LSP cannot reliably
+double-spend the funding transaction.
+In short, the client has the option whether or not to trust a 0-conf
+funding transaction from the LSP.
+If the client does not trust the LSP, the client SHOULD wait for at
+least 1 confirmation and check that the funding transaction output is
+the expected script and value.
+
+In a "client trusts LSP" model, the client MUST transmit the preimage
+as soon as the incoming HTLC has been irrevocably committed.
+The LSP can cheat in this model by giving a random number for the funding
+transaction ID, and the client cannot validate this, as the LSP is
+allowed to not broadcast the funding transaction until the client transmits
+the preimage.
+
+The LSP SHOULD default to "LSP trusts client" model.
+
+The LSP MAY signal a "client trusts LSP" model, and a field is specified
+for the LSP to do this.
+The LSP might do this for example if it detects that it is being attacked
+and needs to protect its funds, even though most of the time it is in
+"LSP trusts client" model.
+
+A client that does not want to trust any LSP can simply refuse to use an
+LSP that signals "client trusts LSP".
+
+LSPs that implement this API SHOULD include mitigations to allow them to
+determine if clients can be trusted, or to detect and protect against attacks
+on the LSP.
+How those mitigations are performed is beyond the scope of this specification.
+
+### Actors
+
+The 'LSP' is the API provider.
+The 'client' is the API client.
+The 'payer' is an entity that pays to the client.
+
+## Flow
+
+Overview:
+
+* The client requests for an SCID it can use in an invoice via
+  `lsps4.reserve_scid`.
+  The client generates the invoice using the given SCID, and issues
+  the invoice to the payer.
+* Payer pays, forwarding towards the LSP with the given SCID as the
+  next hop.
+  When payment arrives at the LSP, it creates pending payment parts
+  (PPPs, a fancy invented term for HODL HTLCs) for it.
+* LSP notifies the client with `lsps4.you_have_new_ppp`, indicating that
```

Comment:
(boring) name suggestion:`lsps4.new_ppp`

Reply:
But that is so boring.... "You have new mail!!!"